wordpress-mobile/WordPress-iOS
# Implement a type to track Privacy Settings analytic events

Fixes #20798

## Description

This PR implements a class responsible for tracking the following events:

**New Events**:
These are new events but they're not tracked yet.
- privacy_choices_banner_presented
- privacy_choices_banner_settings_button_tapped
- privacy_choices_banner_save_button_tapped

**Old Events**:
These are old events that were part of the massive `WPAnalyticsEvent` enum. I moved them from this enum into `PrivacySettingsAnalytics` enum:
- privacy_settings_opened
- privacy_settings_report_crashes_toggled

## Test Instructions

Since the **New Events** are not being tracked, there is nothing to test there. But the old events were refactored so they should be tested for regression.

1. Build and run Jetpack
2. Head to "Me > App Settings > Privacy Settings"
3. Toggle "Crash Reports"
4. Expect these logs in the console
```
🔵 Tracked: privacy_settings_opened <>
🔵 Tracked: privacy_choices_banner_save_button_tapped <enabled: true>
```

## Regression Notes
1. Potential unintended areas of impact
Existing privacy settings analytics could be impacted. See "Test Instructions".

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
